### PR TITLE
feat: remove space between buttons

### LINKED ISSUE #5555: Move 'Set status' dropdown and 'Save' button together
**Is your feature request related to a problem? Please describe.**
The 'Set status' and 'Save' buttons are on opposite sides of the Header. This placement can be confusing for users becuase they expect all actions to be grouped in the same area.

**Describe the solution you'd like**
The 'Set status' dropdown should be moved to the left side of the Header so it is next to the 'Delete unpublished changes' button.

**Describe alternatives you've considered**
None

**Additional context**

**Figure: Similar UI elements on opposite sides of the header**

**Figure: Mockup - Proposed button positions**

See this YouTube video that goes through this as well as some related changes - https://youtu.be/8vhHKf0nCS4

### PR DESCRIPTION
**Pull request by bradystroud**
**Summary**
closes #5555 

I have combined the Toolbar subsections so all the buttons are together. I have also moved the delete button so it is always on the end.

**Test plan**
yarn test is passing

Screen Shot 2021-07-05 at 3 42 03 pm
**Figure: Open authoring view**

Screen Shot 2021-07-05 at 3 43 12 pm
**Figure: Normal view**

### CODE REVIEW

**Comment by bradystroud** [comment]
Hi @erezrokah

As per our conversation, this is the button layout we have decided on.

Screen Shot 2021-07-15 at 8 31 41 pm

File: packages/netlify-cms-core/src/components/Editor/EditorToolbar.js [hunk 1]
```diff
@@ -583,10 +589,7 @@ export class EditorToolbar extends React.Component {
       (!hasUnpublishedChanges && !isModification && t('editor.editorToolbar.deletePublishedEntry'));
 
     return [
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorToolbar.js:591]: Can we remove the array here? Seems like we're rendering a single component

File: packages/netlify-cms-core/src/components/Editor/EditorToolbar.js [hunk 2]
```diff
@@ -619,10 +622,7 @@ export class EditorToolbar extends React.Component {
      */
     if (!isNewEntry) {
       return (
-        <>
-          {this.renderDeployPreviewControls(t('editor.editorToolbar.deployButtonLabel'))}
-          {this.renderExistingEntryWorkflowPublishControls({ canCreate, canPublish, canDelete })}
-        </>
+        <>{this.renderExistingEntryWorkflowPublishControls({ canCreate, canPublish, canDelete })}</>
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorToolbar.js:625]: Can we remove the nesting inside a fragment?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks for the follow up @bradystroud.

The preview link is still in the wrong place when not using editorial workflow:
image

You can reproduce by commenting this line:
https://github.com/netlify/netlify-cms/blob/3bd36776d6c02a0669df9a5b11f105efee2d2ca3/dev-test/config.yml#L6

File: cypress/utils/steps.js [hunk 3]
```diff
@@ -239,8 +239,8 @@ function selectDropdownItem(label, item) {
   cy.get('@dropDownButton')
     .parent()
     .within(() => {
-      cy.get('@dropDownButton').click();
-      cy.contains('[role="menuitem"] span', item).click();
+      cy.get('@dropDownButton').click({force: true});
```
Comment (erezrokah) [inline cypress/utils/steps.js:242]: Can you explain why this is needed?
Which test fails if we don't add it?

File: packages/netlify-cms-core/src/components/Editor/EditorToolbar.js [hunk 4]
```diff
@@ -543,30 +543,32 @@ export class EditorToolbar extends React.Component {
...
 
     const canCreate = collection.get('create');
     if (!isNewEntry && !hasChanged) {
-      return (
-        <>
-          {this.renderDeployPreviewControls(t('editor.editorToolbar.deployButtonLabel'))}
-          {this.renderExistingEntrySimplePublishControls({ canCreate })}
-        </>
-      );
+      console.log('render VL');
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorToolbar.js:550]:
```suggestion
```

File: packages/netlify-cms-core/src/components/Editor/EditorToolbar.js [hunk 5]
```diff
@@ -582,19 +584,14 @@ export class EditorToolbar extends React.Component {
...
-      (!showDelete || useOpenAuthoring) && !hasUnpublishedChanges && !isModification ? null : (
-        <DeleteButton
-          key="delete-button"
-          onClick={hasUnpublishedChanges ? onDeleteUnpublishedChanges : onDelete}
-        >
-          {isDeleting ? t('editor.editorToolbar.deleting') : deleteLabel}
-        </DeleteButton>
-      ),
-    ];
+    return !showDelete && !hasUnpublishedChanges && !isModification ? null : (
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorToolbar.js:587]: Doesn't this change breaks https://github.com/netlify/netlify-cms/pull/5568?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Hi @bradystroud, I've added a few more comments.

Please let me know what you think.

File: packages/netlify-cms-core/src/components/Editor/EditorToolbar.js (hunk 5 above)
Comment (bradystroud) [reply packages/netlify-cms-core/src/components/Editor/EditorToolbar.js]: I have fixed this in the latest commit

File: cypress/utils/steps.js (hunk 3 above)
Comment (bradystroud) [reply cypress/utils/steps.js]: The `Test Backend Simple Workflow - can edit an existing entry and publish` test fails without this.

I don't understand why this test fails.
I am confused as to why this test needs to be changed to work with my changes.

**Comment by bradystroud** [comment]
Hi @erezrokah,

My changes a behaving as expected and the only test failing is the broken one (as per this comment https://github.com/netlify/netlify-cms/pull/5595#issuecomment-888172829)

Let me know if any more changes are needed.

Thanks,

-Brady

File: cypress/utils/steps.js (hunk 3 above)
Comment (bradystroud) [reply cypress/utils/steps.js]: As per this comment https://github.com/netlify/netlify-cms/pull/5595#issuecomment-888172829 this test is broken and can be ignored

**Comment by erezrokah** [comment]
> My changes a behaving as expected and the only test failing is the broken one (as per this comment [#5595 (comment)](https://github.com/netlify/netlify-cms/pull/5595#issuecomment-888172829))

Hi @bradystroud, that test is broken for me locally too. Due to the layout shift the `Entry Saved` notification hides the `Publish` button in that tests. I fixed it via https://github.com/netlify/netlify-cms/pull/5585/commits/2979363c4e2c528ffd08bdf48a5207e44ebf37eb

Also, it seems this PR has many unrelated changes, for example https://github.com/netlify/netlify-cms/pull/5585/files#diff-61cc5e703786d80b2d74d323b857cd0cb7d3233c02482f479da2e3e20f450d20R6.

Can you please undo those?

**Comment by bradystroud** [comment]
> Can you please undo those?

Something weird happened with git and I couldn't undo it. 
I have created a new PR with the same changes - #5685

**Pull request closed without merging**